pymove/tests/core/test_data_frame.py

```diff
@@ -0,0 +1,1102 @@
+        [False, False, False, False],
+        [False, False, False, False],
+        [False, False, False, False],
+    ], columns=['lat', 'lon', 'datetime', 'id'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(move_df.isna(), expected)
+
+
+def test_fillna():
+    move_df = _default_move_df()
+    move_df.at[0, LATITUDE] = nan
+
+    move_test = move_df.fillna(0)
+    expected = DataFrame(data=[
+        [0, 116.319236, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+    ], columns=['lat', 'lon', 'datetime', 'id'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(move_test, expected)
+    assert isinstance(move_test, PandasMoveDataFrame)
+    assert move_df.isna().any(axis=None)
+
+    move_df.fillna(0, inplace=True)
+    assert_frame_equal(move_df, expected)
+
+
+def test_dropna():
+    move_df = _default_move_df()
+    move_df.at[0, LATITUDE] = nan
+
+    move_test = move_df.dropna(axis=1)
+    expected = DataFrame(data=[
+        [116.319236, Timestamp('2008-10-23 05:53:05'), 1],
+        [116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+        [116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+        [116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+    ], columns=['lon', 'datetime', 'id'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(move_test, expected)
+    assert move_test.shape == (4, 3)
+    assert isinstance(move_test, DataFrame)
+    assert move_df.shape == (4, 4)
+
+    move_test = move_df.dropna(axis=0)
+    expected = DataFrame(data=[
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+    ], columns=['lat', 'lon', 'datetime', 'id'], index=[1, 2, 3]
+    )
+    assert_frame_equal(move_test, expected)
+    assert move_test.shape == (3, 4)
+    assert isinstance(move_test, PandasMoveDataFrame)
+    assert move_df.shape == (4, 4)
+
+    try:
+        move_df.dropna(axis=1, inplace=True)
+        assert False
+    except AttributeError:
+        assert True
+
+    move_df.dropna(axis=0, inplace=True)
+    assert_frame_equal(move_df, expected)
+
+
+def test_sample():
+    move_df = _default_move_df()
+
+    sample_test = move_df.sample(n=2, random_state=42)
+    expected = DataFrame(data=[
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+    ], columns=['lat', 'lon', 'datetime', 'id'], index=[1, 3]
+    )
+    assert_frame_equal(sample_test, expected)
+    assert isinstance(sample_test, PandasMoveDataFrame)
+
+
+def test_isin():
+    move_df = _default_move_df()
+    move_df_copy = _default_move_df()
+
+    assert move_df.isin(move_df_copy).all(axis=None)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574862)